### PR TITLE
V1.2-delete-tasks

### PR DESCRIPTION
**Pull request by amresh2000**
- Add command to delete a task: `DeleteTaskCommand`
- Add tests for new class

### CODE REVIEW

**Comment by wxy1203** [comment]
DeleteTaskCommandTest.java:49: no newline at EOF.
Make a new line at the end of file to fix CI error.

File: src/main/java/seedu/address/logic/commands/task/DeleteTaskCommand.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+public class DeleteTaskCommand extends TaskCommand {
+    public static final String COMMAND_WORD = "delete";
+    public static final String COMMAND_WORD_FULL = TaskCommand.COMMAND_WORD + " " + COMMAND_WORD;
+    public static final String MESSAGE_USAGE = COMMAND_WORD_FULL
+            + ": Deletes the task that index number refers to.\n"
+            + "Parameters: \n"
+            + "Example: " + COMMAND_WORD + " 1";
+    public static final String MESSAGE_SUCCESS = "Task deleted: %1$s";
+    public static final String MESSAGE_DELETE_TASK_SUCCESS = "Deleted Task: %1$s";
```
Comment (kianseong) [inline src/main/java/seedu/address/logic/commands/task/DeleteTaskCommand.java:24]: Do we need two success messages?

File: src/test/java/seedu/address/logic/commands/task/AddTaskCommandTest.java [hunk 2]
```diff
@@ -81,7 +81,7 @@ public void equals() {
     /**
      * A default model stub that have all the methods failing.
      */
-    private class ModelStub implements Model {
+    protected class ModelStub implements Model {
```
Comment (kianseong) [inline src/test/java/seedu/address/logic/commands/task/AddTaskCommandTest.java:84]: Why are we increasing the visibility?

File: src/test/java/seedu/address/logic/commands/task/AddTaskCommandTest.java [hunk 3]
```diff
@@ -199,7 +204,7 @@ public boolean hasTask(Task task) {
     /**
      * A Model stub that always accept the person being added.
      */
-    private class ModelStubAcceptingTaskAdded extends ModelStub {
+    protected class ModelStubAcceptingTaskAdded extends ModelStub {
```
Comment (kianseong) [inline src/test/java/seedu/address/logic/commands/task/AddTaskCommandTest.java:207]: JavaDoc and name not matching.

File: src/test/java/seedu/address/logic/commands/task/DeleteTaskCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
+package seedu.address.logic.commands.task;
+
+import org.junit.jupiter.api.Test;
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.*;
```
Comment (kianseong) [inline src/test/java/seedu/address/logic/commands/task/DeleteTaskCommandTest.java:8]: Should we be using wildcard imports?

File: src/test/java/seedu/address/logic/commands/task/DeleteTaskCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+import seedu.address.model.task.Task;
+import seedu.address.testutil.TaskBuilder;
+import seedu.address.testutil.TaskPanelBuilder;
+
+import java.util.Arrays;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (kianseong) [inline src/test/java/seedu/address/logic/commands/task/DeleteTaskCommandTest.java:15]: Similarly, we should not be using wildcard imports.

**Review by kianseong — COMMENTED** [review]
Looks good albeit some minor discrepancies.

**Pull request closed without merging**